### PR TITLE
[WEB-28] [EDITOR] implement code blocks

### PR DESCRIPTION
**Pull request by josh-ramos-22**
### Why are the changes required?
Currently, there is no way to style text as `code`.
Being a society for Computer Science, CSESoc articles will inevitably need to have some integrated `code` styling.

### Changes
- New file `frontend/src/cse-ui-kit/small_buttons/CodeButton.tsx`: contains button component used to toggle code styling.
- New SVG file `frontend/src/cse-ui-kit/assets/code-button.svg` for the `CodeButton` component
- `frontend/src/packages/editor/components/EditorBlock.tsx`
  - added code styles required in the `Text` styled component
- New file `frontend/src/packages/editor/components/buttons/EditorCodeButton.tsx`: Defines code button in editor used to toggle code styling
- `frontend/src/packages/editor/components/buttons/buttonHelpers.ts`
  -  Added code to denote `code` as a valid text decoration
  - added event handler case to enable `` Ctrl ` `` as the keyboard shortcut for code style toggling.
 - `frontend/src/packages/editor/types.ts`
   - added `code` to the type definition of `CustomText`

### Screenshots
Screenshot 2023-04-05 205214

### CODE REVIEW

File: frontend/src/packages/editor/components/buttons/EditorCodeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { toggleMark } from "./buttonHelpers";
+
+const EditorCodeButton: FC = () => {
+  const editor = useSlate();
+  return (
+    <CodeButton
+      size={30}
+      onMouseDown={(event) => {
+        event.preventDefault();
+        // TODO switch mark to code (not sure if this has to be defined)
```
Comment (lauraw0) [inline frontend/src/packages/editor/components/buttons/EditorCodeButton.tsx:13]: i believe we toggle mark as it helps present the array of the text in database with a mark that the following contents is a inline code block. We then use this mark in the homepage to display that text as code. Feel free to ask me about this in more depth if this doesn't make any sense :D
e.g. `[{"align":"left","quote":true,"underline":true,"code":true,"text":"sndfgjsbdfjg"}]}]`

Comment (lauraw0) [reply frontend/src/packages/editor/components/buttons/EditorCodeButton.tsx:13]: once it is clear feel free to remove the comment and merge into main :)

**Review by lauraw0 — APPROVED** [review]
lgtm! pls check comments and then merge whenever 👍

File: frontend/src/packages/editor/components/buttons/EditorCodeButton.tsx (hunk 1 above)
Comment (josh-ramos-22) [reply frontend/src/packages/editor/components/buttons/EditorCodeButton.tsx]: all good! just forgot to delete the comment

**Pull request merged**